### PR TITLE
Omf fixups

### LINKED ISSUE #212158: oh-my-fish: omf-install fails
### Describe the bug
omf-install fails because it copies read-only files and directories from the Nix store (and the resulting installation cannot work without modifying the permissions)

### Steps To Reproduce
1. omf-install

### Expected behavior
The installation should suceed.

### Screenshots
```
pierre@force ~> omf-install
Checking for a sane environment...
Installing Oh My Fish to /home/pierre/.local/share/omf...
Writing bootstrap to /home/pierre/.config/fish/conf.d/omf.fish...
Setting up Oh My Fish configuration...
Updating https://github.com/oh-my-fish/packages-main master... Done!
Installing package default
fatal: could not create work tree dir '/home/pierre/.local/share/omf/pkg/default': Permission denied
Could not install default.
Install aborted: Error installing plugins

Oh My Fish installation failed
```

### Additional context
 ~/.local/share/omf/ files are all without write permission

### Notify maintainers
@AndersonTorres 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-linux"`
 - host os: `Linux 5.10.153-20432-gaa06ea936644, Debian GNU/Linux, 11 (bullseye), nobuild`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.1`
 - channels(pierre): `"nixpkgs"`
 - nixpkgs: `/home/pierre/.nix-defexpr/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/212158

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -11539,8 +11539,6 @@ with pkgs;
 
   oh-my-posh = callPackage ../development/tools/oh-my-posh { };
 
-  oh-my-zsh = callPackage ../shells/zsh/oh-my-zsh { };
-
```
Comment (mweinelt) [inline pkgs/top-level/all-packages.nix:11543]: This unrelated top-level attribute was removed.

**Comment by gbirke** [comment]
Thank you for the fixup!
I managed to install oh-my-fish on NixOS 23.11 (from unstable), but not without a hitch: the `@OMF@` placeholder in the `omf-install` "binary" is not replaced with the path to the file in the nix store. I manually executed the installer command, replacing the placeholder with my nix store path. After the installation everything is in ~/.local/share/omf anyway, so it's not a big problem.

I don't know enough about Nix to have any helpful suggestions on how to fix this or if it's just a case of installing a package from unstable, just wanted to point out that you might need a followup PR for this

**Comment by hugolgst** [comment]
Same issue with oh-my-fish.

**Comment by hugolgst** [comment]
```
error: Error reading script file '@OMF@/share/oh-my-fish/bin/install':
error: No such file or directory
```

**Comment by hugolgst** [comment]
how was this merged?

**Comment by AndersonTorres** [comment]
> how was this merged?

No idea...

**Pull request merged**